src/middlewared/middlewared/plugins/pool.py

```diff
@@ -525,6 +535,21 @@ def pool_extend(self, pool):
         Str('name', required=True),
         Bool('encryption', default=False),
         Str('deduplication', enum=[None, 'ON', 'VERIFY', 'OFF'], default=None, null=True),
+        Dict(
+            'encryption_options',
+            Bool('generate_key', default=False),
+            Bool('key_file', default=False),
+            Int('pbkdf2iters', default=350000, validators=[Range(min=100000)]),
+            Str(
+                'algorithm', default='AES-256-CCM', enum=[
+                    'AES-128-CCM', 'AES-192-CCM', 'AES-256-CCM', 'AES-128-GCM', 'AES-192-GCM', 'AES-256-GCM'
+                ]
+            ),
+            Str('key_format', default='KEY', enum=[ZFSKeyFormat.PASSPHRASE.value, 'KEY']),
```

Comment:
key_format = KEY? is that intended?

Reply:
Also, we already have `key` and `passphrase` attributes, do we really need `key_format` ?